### DISCUSSION TITLE
Form field content isn't updating

### DISCUSSION THREAD

**Discussion by bmaupin** [remix-run/remix#1797] [Q&A] [answered]
I'm completely new to Remix. I'm following the [blog tutorial](https://remix.run/docs/en/v1/tutorials/blog) and trying to do the "homework" portion at the end, which is to create an edit page for the posts, which I've done here: https://github.com/bmaupin/remix-test/blob/main/app/routes/admin/%24slug.tsx

It works, but I'm seeing weird behaviour:

- When I first browse the page to edit a post, if I switch to another post to edit, the form field content doesn't change to reflect that post's content
- If I reload the page, the post's content shows up properly, but then when I switch to a different post, only the title and slug update, not the markdown
- If I disable JavaScript, everything works as expected
- I put a `console.log` in `EditPost` that only gets called either when I refresh the page or when I disable JavaScript

You can see it for yourself here: https://bmaupin-remix-test.herokuapp.com/admin

I tested in Chrome and Firefox with two different hosting providers, and I see the same behaviour when I'm running it locally.

My guess is I've done something wrong but I'm not sure what.

Thanks!

**Comment by sergiodxa** [comment, marked as answer]
The problem is you are using defaultValue to set the values from the loader, but because the transition happens with JS React is ignoring changes in those values, what you can do is to use the useLocation to get the `location.key` and force a re-mount of the inputs after a navigation.

```ts
let { initialValue } = useLoaderData()
let location = useLocation()
return <input key={location.key} name="something" defaultValue={initialValue} />
```

That way, after a route transition the `location.key` will change and React will re-mount the input using the new defaultValue.

You can put the key in the Form to force a re-mount of everything.

**Comment by bmaupin** [comment]
@sergiodxa I'm guessing you meant this? There doesn't seem to be a key property on the transition.

```javascript
const location = useLocation();
return <input key={location.key} name="something" defaultValue={initialValue} />
```

That's what I did (well, I did it on the whole Form because I'm being lazy: https://github.com/bmaupin/remix-test/commit/185b6bfcf76425dad0db18f6b0eee77aaf1184e5) and it worked. Thanks so much!

**Reply by sergiodxa** [reply]
Yes, I meant location, I fixed my comments, sorry for the confusion 🙏
